Documentation/contributing/docs/docsframework.rst

```diff
@@ -0,0 +1,382 @@
+with lowercase exclusively, or uppercase exclusively, are case-insensitive
+exceptions for spell-checking. Words with mixed case are case-sensitive. This
+file should remain sorted alphabetically.
+
+To add new entries to the list, run ``Documentation/update-spelling_wordlist.sh``.
+
+To clean-up obsolete entries, make sure the spell checker reports no issue on
+the current version of the documentation, then remove all entries from the
+file, run the spell checker and re-add all reported exceptions.
+
+We use a custom filter for the spell checker, for spelling ``WireGuard``
+correctly as ``WireGuard``, or ``wireguard`` in some contexts, but never as
+``Wireguard``. This filter is implemented in
+``Documentation/_exts/cilium_spellfilters.py`` and registered in
+``Documentation/conf.py``.
+
+.. _spell-checker module: https://github.com/sphinx-contrib/spelling
+.. _builder: https://www.sphinx-doc.org/en/master/usage/builders
+
+:spelling:ignore:`rstcheck`
+---------------------------
+
+We use `rstcheck`_ to validate the rST formatting. We have a list of warnings
+to ignore (in part because the linter has bugs). The way we call the tool, and
+this list of exceptions, are configured in ``Documentation/check-build.sh``.
+
+.. _rstcheck: https://rstcheck.readthedocs.io
+
+Link checker
+------------
+
+We have a link checker under ``Documentation/check-links.sh`` for the
+documentation. However, due to some unsolved issues, we do not run it in CI.
+See :gh-issue:`27116` for details.
+
+Web server for local preview
+----------------------------
+
+Launch a web server to preview the generated documentation locally with ``make
+render-docs``.
+
+See more on this topic on the page about :ref:`testing documentation
+<testing-documentation>` page.
+
+Custom Sphinx roles
+-------------------
+
+We use several custom roles in the documentation:
+
+- ``git-tree``
+- ``github-project``
+- ``github-backport``
+- ``gh-issue``
+- ``prev-docs``
+
+We use these roles to insert links based on specific URL templates, via the
+`extlinks`_ extension. They are all configured in ``Documentation/conf.py``.
+They should be used wherever relevant, to ensure that the way we link to
+different resources remains consistent.
+
+.. _extlinks: https://www.sphinx-doc.org/en/master/usage/extensions/extlinks.html
+
+Custom Sphinx directives
+------------------------
+
+We do not implement custom directives as of this writing.
+
+Custom extensions
+-----------------
+
+We use custom extensions for Sphinx, implemented under ``Documentation/_exts``.
+
+- One defines the custom filters for the spell checker.
+- One patches Sphinx's HTML translator to open all external links in new tabs.
+
+Google Analytics
+----------------
+
+The documentation uses Google Analytics to collect metrics. This is configured
+in ``Documentation/conf.py``.
+
+Customization
+-------------
+
+Additional elements of customization for Cilium's documentation and defined in
+the main repository include:
```

Comment:
Clarity nit
```suggestion
Additional elements of customization for Cilium's documentation are defined in
the main repository:
```

Reply:
Not all of them are defined in that repository. The theme, in particular, has its own repository.